include/ozo/error.h

```diff
@@ -731,4 +822,141 @@ inline const error_category& category() noexcept {
 
 } // namespace sqlstate
 
+namespace errc {
+
+inline auto make_error_condition(const code e) {
+    return error_condition(static_cast<int>(e), category());
+}
+
+namespace impl {
+
+namespace asio = boost::asio;
+namespace hana = boost::hana;
+
+template <code Code>
+struct codes_for_condition;
+
+template <>
+struct codes_for_condition<connection_error> {
+    constexpr static auto value = hana::make_tuple(
+        // SQL state
+        ozo::sqlstate::connection_exception,
+        // Asio errors
+        asio::error::broken_pipe,
+        asio::error::connection_aborted,
+        asio::error::connection_refused,
+        asio::error::connection_reset,
+        asio::error::fault,
+        asio::error::host_unreachable,
+        asio::error::interrupted,
+        asio::error::network_reset,
+        asio::error::not_connected,
+        asio::error::operation_aborted,
+        asio::error::shut_down,
+        asio::error::timed_out,
+        asio::error::try_again,
+        // Asio netdb_category
+        asio::error::host_not_found,
+        asio::error::host_not_found_try_again,
+        asio::error::no_data,
+        asio::error::no_recovery,
+        // System IO-error
+        boost::system::errc::io_error,
+        // OZO libpq-related errors
+        ozo::error::pq_connection_start_failed,
+        ozo::error::pq_socket_failed,
+        ozo::error::pq_connection_status_bad,
+        ozo::error::pq_connect_poll_failed,
+        ozo::error::pg_send_query_params_failed,
+        ozo::error::pg_consume_input_failed,
+        ozo::error::pg_set_nonblocking_failed,
+        ozo::error::pg_flush_failed
+    );
+};
+
+template <>
+struct codes_for_condition<database_readonly> {
+    constexpr static auto value = hana::make_tuple(ozo::sqlstate::read_only_sql_transaction);
+};
+
+template <>
+struct codes_for_condition<introspection_error> {
+    constexpr static auto value = hana::make_tuple(
+        ozo::error::bad_result_process,
+        ozo::error::bad_object_size,
+        ozo::error::bad_array_size,
+        ozo::error::bad_array_dimension,
+        ozo::error::bad_composite_size,
+        ozo::error::unexpected_eof
+    );
+};
+
+template <>
+struct codes_for_condition<type_mismatch> {
```

Comment:
Type mismatch could occur in case of database schema changed and the types get new OIDs, so in this case, simple reconnect should help and this error condition can be used to detect such a situation.